python_src/src/lamp_py/migrations/versions/performance_manager/695ef31623ae_rt_trunk_id_and_route_id_processing.py

```diff
@@ -0,0 +1,308 @@
+                    NEW.branch_route_id := 'Red-Braintree';
+                ELSEIF rt_red_is_ashmont_branch(NEW.trip_hash) THEN
+                    NEW.branch_route_id := 'Red-Ashmont';
+                ELSE
+                    NEW.branch_route_id := null;
+                END IF;
+                NEW.trunk_route_id := NEW.route_id;
+            ELSE
+                NEW.branch_route_id := null;
+                NEW.trunk_route_id := NEW.route_id;
+            END IF;
+            RETURN NEW;
+        END;
+        $$ LANGUAGE plpgsql;
+    """
+    op.execute(create_get_rt_branch_trunk_id)
+
+    create_trigger = """
+        CREATE TRIGGER rt_trips_update_branch_trunk BEFORE INSERT OR UPDATE ON vehicle_trips 
+        FOR EACH ROW EXECUTE PROCEDURE update_rt_branch_trunk_id();
+    """
+    op.execute(create_trigger)
+
+    update_static_ashmont_func = """
+        CREATE OR REPLACE FUNCTION red_is_ashmont_branch(p_trip_id varchar, p_fk_ts int) RETURNS boolean AS $$ 
+        DECLARE
+            is_ashmont boolean;
+        BEGIN 
+            SELECT
+                true
+            INTO
+                is_ashmont
+            FROM 
+                static_stop_times sst
+            WHERE 
+                sst.trip_id = p_trip_id
+                AND sst."timestamp" = p_fk_ts
+                AND sst.stop_id IN ('70087', '70088', '70089', '70090', '70091', '70092', '70093', '70094')
+            LIMIT 1
+            ;
+
+            IF FOUND THEN
+                RETURN true;
+            ELSE
+                RETURN false;
+            END IF;
+        END;
+        $$ LANGUAGE plpgsql;
+    """
+    op.execute(update_static_ashmont_func)
+
+    update_static_braintree_func = """
+        CREATE OR REPLACE FUNCTION red_is_braintree_branch(p_trip_id varchar, p_fk_ts int) RETURNS boolean AS $$ 
+        DECLARE	
+            is_braintree boolean;
+        BEGIN 
+            SELECT
+                true
+            INTO
+                is_braintree
+            FROM 
+                static_stop_times sst
+            WHERE 
+                sst.trip_id = p_trip_id
+                AND sst."timestamp" = p_fk_ts
+                AND sst.stop_id IN ('70097', '70098', '70099', '70100', '70101', '70102', '70103', '70104', '70105')
+            LIMIT 1
+            ;
+
+            IF FOUND THEN
+                RETURN true;
+            ELSE
+                RETURN false;
+            END IF;
+        END;
+        $$ LANGUAGE plpgsql;
+    """
+    op.execute(update_static_braintree_func)
+
+    update_view = """
+        CREATE OR REPLACE VIEW static_service_id_lookup AS 
+        WITH mod_feed_dates AS (
+            SELECT 
+                feed_start_date::text::date AS start_date,
+                feed_end_date::text::date AS end_date,
+                "timestamp"
+            FROM public.static_feed_info
+        ), sd_match AS (
+            SELECT 
+                replace(mod_sd::date::text, '-', '')::integer AS service_date,
```

Comment:
This view is very similar to the one created for Heather, but provides all service dates available based on `feed_start_date` and `feed_end_date` values.